lib/python/rose/apps/fcm_make.py

```diff
@@ -45,61 +47,90 @@ def run(self, app_runner, conf_tree, opts, args, uuid, work_files):
-            target = auth + ":"
-            if use_pwd:
-                target += os.path.join(t.suite_dir_rel, "work", t.task_id)
-            else:
-                target += os.path.join(t.suite_dir_rel, "share", t.task_name)
-            env_export("ROSE_TASK_MIRROR_TARGET", target,
-                       app_runner.event_handler)
-            # N.B. MIRROR_TARGET deprecated
-            env_export("MIRROR_TARGET", target, app_runner.event_handler)
+        if task_name_orig2cont[CONT] in task.task_name:
+            return self._run_cont(
+                app_runner, conf_tree, args, task, task_name_orig2cont)
 
         cmd = ["fcm", "make"]
-        for cfg in ["fcm-make.cfg",
-                    os.path.join(t.suite_dir, "etc", t.task_name + ".cfg")]:
+        use_pwd = conf_tree.node.get_value(["use-pwd"]) in ["True", "true"]
+        for cfg in [
+                "fcm-make.cfg",
+                os.path.join(task.suite_dir, "etc", task.task_name + ".cfg")]:
             if os.access(cfg, os.F_OK | os.R_OK):
                 if cfg != "fcm-make.cfg" or not use_pwd:
                     cmd += ["-f", os.path.abspath(cfg)]
                 break
         if not use_pwd:
-            cmd += ["-C", os.path.join(t.suite_dir, "share", t.task_name)]
-        cmd_opt_jobs = conf_tree.node.get_value(
-                ["opt.jobs"], os.getenv("ROSE_TASK_N_JOBS", self.OPT_JOBS))
-        cmd += ["-j", cmd_opt_jobs]
-        cmd_args = conf_tree.node.get_value(["args"],
-                                            os.getenv("ROSE_TASK_OPTIONS"))
+            cmd += [
+                "-C",
+                os.path.join(task.suite_dir, "share", task.task_name)]
+        cmd += ["-j", conf_tree.node.get_value(
+            ["opt.jobs"], os.getenv("ROSE_TASK_N_JOBS", self.OPT_JOBS))]
+        cmd_args = conf_tree.node.get_value(
+            ["args"], os.getenv("ROSE_TASK_OPTIONS"))
         if cmd_args:
             cmd += shlex.split(cmd_args)
         if args:
             cmd += args
+
+        # "mirror" for backward compat. Use can specify a null string as value
+        # to switch off the mirror target configuration.
+        task_name_cont = task.task_name.replace(
+            task_name_orig2cont[ORIG], task_name_orig2cont[CONT])
+        auth = app_runner.suite_engine_proc.get_task_auth(
+            task.suite_name, task_name_cont)
+        if auth is not None:
+            target = auth + ":"
+            if use_pwd:
+                target += os.path.join(
+                    task.suite_dir_rel, "work", task.task_cycle_time,
+                    task_name_cont)
+            else:
+                target += os.path.join(
+                    task.suite_dir_rel, "share", task.task_name)
+            # Environment variables for backward compat. "fcm make"
+            # supports arguments as extra configurations since version
+            # 2014-03.
+            for name in ["ROSE_TASK_MIRROR_TARGET", "MIRROR_TARGET"]:
```

Comment:
Yes, but this is not a change to functionality for the `fcm_make` built-in application.